doc/secure_boot/Secureboot_uefi_key_management.md

```diff
@@ -0,0 +1,217 @@
+2. Use CLI to send the authenticated variable to UEFI firmware.
+3. UEFI firmware checks the trust source and integrity of authenticated variables and enrolls into UEFI database.
+4. The system requires a reboot to activate the keys for use.
+
+### 4.3 Remove keys from UEFI database
+For removal of keys requires the removal of an authenticated variable outside of the system. 
+
+````
+Sample example of authenticated variable creation for removal of the key from the signature database list:
+E.g The UEFI database has 3 entries of keys and one of them to be removed here
+
+1. Read the db key list from UEFI DB. 
+   efi-readvar -v Db -o DB-orig.esl 
+2. Convert signature list to individual certs. This command extracts DB-orig-1.der DB-orig-2.der DB-orig-3.der so on key files
+   sig-list-to-certs DB-orig.esl DB-orig
+3. Convert each of the certs into pem format
+   openssl x509 -inform der -in db-orig-1.der -out db-orig-1.pem
+   openssl x509 -inform der -in db-orig-2.der -out db-orig-2.pem
+   openssl x509 -inform der -in db-orig-3.der -out db-orig-3.pem
+4. Removal of db-orig-2.pem to be discarded
+5. Convert db-orig-1.pem and db-orig-3.pem to esl format
+   cert-to-efi-sig-list -g <GUID> db-orig-1.pem db-orig-1.esl
+   cert-to-efi-sig-list -g <GUID> db-orig-3.pem db-orig-3.esl
+6. Concatenate these two esl files 
+   cat db-orig-1.esl db-orig-3.esl > DB-new.esl
+7. Create authenticated variable and sign it
+   sign-efi-sig-list -g "<GUID>" -k KEK.key -c KEK.crt db DB-new.esl DB-new.auth
+8. Copy DB-new.auth to the system and use CLI to update UEFI database
+
+````
+
+### 4.4 Revoke keys
+Revoking a key from the UEFI database is done when the previously trusted key is no longer trustworthy. This action is taken when a key's security has been compromised.
+It is achieved by inserting the authenticated variable for the compromised key into the forbidden key list(DBx). 
+
+```
+Sample example:
+If db-orig-1.pem is compromised the generate an authenticated variable for db-orig-1.pem
+1. Convert ESL of pem cert.
+   cert-to-efi-sig-list -g <GUID> db-orig-3.pem db-orig-3.esl
+2. Create authenticated variable
+   sign-efi-sig-list -g "<GUID>" -k KEK.key -c KEK.crt dbx db-orig-3.esl DB-revoke.auth
+3. Copy DB-revoke.auth to the system and use CLI to update the UEFI database
+```
+
+### 4.5 Show keys
+This is to display the PK, KEK, DB, and DBx key lists.
+
+## 5 SONIC implementation
+The access of UEFI variables may vary from platform to platform depending on the specific implementation of UEFI firmware. Sonic will provide a common layer for accessing these variables and vendors can implement platform-specific parts. The SONIC will provide a plugin Python for vendors to implement these APIs and integrate them with the SONiC generic layer.
+
+### 5.1 Plugin class
+
+![image](https://github.com/sacnaik/SONiC/assets/25231205/a42cd039-0675-4a71-84e0-bce00eb2ca9d)
+
+### 5.2 OVMF based platform can use Linux implementation to access UEFI variable
```

Comment:
can you clarify the cisco code contribution here?

Reply:
The HLD is referred to as OVMF since it is Open firmware and available for someone to experiment with. Cisco BIOS has implemented a similar approach to OVMF.  When Cisco8000 implements SONiC UEFI key management for Cisco hardware similar code will be added under Cisco platform-specific code.  That means the common contribution here is the SONiC CLI and Python module defining the base class that is mentioned in the document.